src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreventNumericIntPtrUIntPtrBehavioralChangesTests.cs

```diff
@@ -0,0 +1,283 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.CSharp.Analyzers.Runtime.CSharpPreventNumericIntPtrUIntPtrBehavioralChanges,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class PreventNumericIntPtrUIntPtrBehavioralChangesTests
+    {
+
+        [Fact]
+        public async Task IntPtrAdditionSubstructionWithFieldReference()
+        {
+            await PopulateTestCs(@"
+using System;
+
+class Program
+{
+    IntPtr intPtr1;
+    IntPtr intPtr2;
+
+    public void M1()
+    {
+        checked
+        {
+            int i = 0;
+            intPtr2 = {|#0:intPtr1 + 2|}; // Built in operator '+' now may throw when overflowing in checked context. Wrap the expression with 'unchecked' statement to restore old behavior.
+            intPtr2 = {|#1:intPtr1 - 2 * 3|}; // Built in operator '-' now may throw when overflowing in checked context. Wrap the expression with 'unchecked' statement to restore old behavior.
+            intPtr2 = {|#2:intPtr1 - 2|} - 3; // Built in operator '-' now may throw when overflowing in checked context. Wrap the expression with 'unchecked' statement to restore old behavior.
+            intPtr2 = {|#3:intPtr1 + i|} + 3; // Built in operator '+' now may throw when overflowing in checked context. Wrap the expression with 'unchecked' statement to restore old behavior.
```

Comment:
Perhaps nit-picky...

It looks like these diagnostics are expected to be `OperatorThrowsRule`, which has the "may or may not" verbiage. The comments reference a different diagnostic message though.

_(I noticed this while scanning to ensure there were tests for each diagnostic that could surface)._

Reply:
Not sure why you thought so, "may or may not" verbiage is used only in the `description`, which is shared between the 3 different `messages`:
 - Built in operator '{0}' now may not throw when overflowing in unchecked context. Wrap the expression with 'checked' statement to restore old behavior.
  - Built in operator '{0}' now may throw when overflowing in checked context. Wrap the expression with 'unchecked' statement to restore old behavior.
  - Built in explicit conversion '{0}' now may throw when overflowing in checked context. Wrap the expression with 'unchecked' statement to restore old behavior.

Reply:
Now I see that the 1st message is wrong it should be 
- Built explicit conversion '{0}' now may not throw when overflowing in unchecked context. Wrap the expression with 'checked' statement to restore old behavior. 
- ~~Built in operator '{0}' now may not throw when overflowing in unchecked context. Wrap the expression with 'checked' statement to restore old behavior~~.